x-pack/legacy/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -192,8 +199,11 @@ export class TaskRunner {
     }
 
     return {
-      alertTypeState: updatedAlertTypeState,
-      alertInstances: instancesWithScheduledActions,
+      alertTypeState: updatedAlertTypeState || undefined,
```

Comment:
ah, interesting.
This isn't actually a change, as it was a `void` before, which is an `undefined` from a function that doesn't return anything.
This has only now become an issue because we're actually typing this exchange!

Any idea what the expected behaviour would be? Would an AlertExecutor that returns no state be expected to wipe out previous state or keep what was there before?

I might split this possible change off of this PR as it feels like it's worth a discussion on its own.
As it stands, an executor won't be able to return `null` as we haven't marked that as a valid return type (unless they aren't using TS... 🤦‍♂ ) so feels like a separate conversation.